src/mono/wasm/runtime/loader/exit.ts

```diff
@@ -6,6 +6,12 @@ import { mono_log_debug, consoleWebSocket, mono_log_error, mono_log_info_no_pref
 
 export function abort_startup(reason: any, should_exit: boolean): void {
     mono_log_debug("abort_startup");
+    // FIXME: Should we set this after rejecting the promises?
```

Comment:
I think this place is OK. You can remove the comment, I don't expect that somebody would try to recover from the rejections trowed here.

We also need to set `runtimeExited` in `mono_exit` method.

Reply:
Originally I was setting it in mono_exit, but it caused calls in to stop working after main(), which breaks a bunch of the samples

Reply:
Hmm.
We have `dotnet.run()` API implemented as `runMainAndExit` and I guess that's why.
It makes sense in console apps, but not so much in browser apps. 

So perhaps web apps should not call `runMainAndExit` but just `runMain`. But that would break our (unit test) expectations about exit code of web apps.

@maraf what do you think ?

Reply:
I think the exit code is also important since a startup failure usually produces a nonzero exit code

Reply:
I think that even web apps should call `runMainAndExit` and after that the runtime should be in the "don't touch me state". Web apps and probably any "interactive apps" should have async main that exits when the app done.

Reply:
Would this be compatible with Blazor and Uno?
@jeromelaban

Reply:
I think Blazor has async main which never resolves the Task, right @maraf  ?

Reply:
@kg for Uno, we are using the `runMain` API to start the apps to call in a synchronous main that exits right away. If runMainAndExit implies that apps need to have an async main, this will be a breaking change. 

I tend to think that the idea of the `Main` method handling exit in the browser does not make sense synchronously nor asynchronously, as terminating the app is not really possible (aside from closing the page/tab).

Reply:
There is `Module.noExitRuntime` in the emscripten which it true in the browser. Perhaps we should set this new `set_runtime_aborted` only in shell ? Or when `Module.noExitRuntime == false`. 

I think think that we also want to set emscripten's `ABORT` flag by calling `_proc_exit()` or `abort()` js functions. At least on current thread.

Also we should register `set_runtime_aborted` into `Module.onAbort` callback.